StRoot/StFwdTrackMaker/include/Tracker/FwdTracker.h

```diff
@@ -0,0 +1,1085 @@
+#ifndef FWD_TRACKER_H
+#define FWD_TRACKER_H
+
+#include "Fit/Fitter.h"
+#include "TFile.h"
+#include "TGraph2D.h"
+#include "TH1.h"
+#include "TH1F.h"
+#include "TH2F.h"
+#include "TRandom3.h"
+#include "TTree.h"
+#include "TVector3.h"
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "StFwdTrackMaker/include/Tracker/ConfigUtil.h"
+#include "StFwdTrackMaker/include/Tracker/FwdHit.h"
+#include "StFwdTrackMaker/include/Tracker/HitLoader.h"
+#include "StFwdTrackMaker/include/Tracker/QualityPlotter.h"
+#include "StFwdTrackMaker/include/Tracker/TrackFitter.h"
+#include "StFwdTrackMaker/include/Tracker/loguru.h"
+
+#include "Criteria/Criteria.h"
+#include "Criteria/ICriterion.h"
+#include "KiTrack/Automaton.h"
+#include "KiTrack/SegmentBuilder.h"
+#include "KiTrack/SubsetHopfieldNN.h"
+
+#include "CriteriaKeeper.h"
+
+#include "GenFit/FitStatus.h"
+
+#include "StFwdTrackMaker/XmlConfig/XmlConfig.h"
+
+// Utility class for evaluating ID and QA truth
+struct MCTruthUtils {
+
+    static int domCon(std::vector<KiTrack::IHit *> hits, float &qual) {
+        std::map<int, int> count;
+        int total = 0;
+
+        for (auto *h : hits) {
+            auto *hit = dynamic_cast<FwdHit *>(h);
+            if (0 == hit)
```

Comment:
Isn't your code who adds hits into the container? If so, you can guarantee to have only pointers to valid objects and avoid the `if (!hit)` checks here and elsewhere. It might be even better if you have copies of the hits in the container.